tests/lac/diagonal_matrix.cc

```diff
@@ -0,0 +1,94 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2016 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE at
+// the top level of the deal.II distribution.
+//
+// ---------------------------------------------------------------------
+
+
+// Tests DiagonalMatrix
```

Comment:
could you please add one more which would be closer to the intended usage in assembly, i.e. compare `SparseMatrix<double>` to `DiagonalMatrix` for `FE_Q(2)` and triangulation with hanging nodes? (laplace or mass matrix, does not matter).